File: testing/backend/integration/integration-tests/src/main/java/com/timsmithsoftware/integration_tests/SystemConfiguration.java
package com.timsmithsoftware.integration_tests;

import java.util.concurrent.CompletableFuture;

import org.junit.runner.JUnitCore;
import org.junit.runner.Result;

import com.timsmithsoftware.integration_tests.models.Connection;
import com.timsmithsoftware.integration_tests.tests.GetVisitsTests;

public class SystemConfiguration {
	Connection[] connections;
	
	public CompletableFuture<Boolean> waitForConnections() {
		return CompletableFuture.completedFuture(false);
		}
	
	public void runTests() {
		JUnitCore junit = new JUnitCore();
		Result result = junit.run(GetVisitsTests.class);
	}

}

File: testing/backend/integration/integration-tests/src/main/java/com/timsmithsoftware/integration_tests/SystemConfiguration.kt
package com.timsmithsoftware.integration_tests

import com.timsmithsoftware.integration_tests.models.ApiConnection
import com.timsmithsoftware.integration_tests.models.Connection
import com.timsmithsoftware.integration_tests.models.DatabaseConnection
import java.net.http.HttpClient

class SystemConfiguration {
    private val httpClient: HttpClient = HttpClient.newHttpClient()
    private val connections: List<Connection> = listOf(
            DatabaseConnection(),
            ApiConnection(httpClient)
    )

    fun waitForConnections(): Boolean {
        for(connection in connections) {
            connection.waitUntilAlive()
        }
        return true
    }
}